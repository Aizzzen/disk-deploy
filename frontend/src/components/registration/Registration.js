import React, {useState} from 'react';
import './Authorization.scss';
import Input from "../../utils/input/Input";
import {registration} from "../../actions/user";

const Registration = () => {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')

    return (
        <div data-testid='register-page' className='authorization'>
            <div className="authorization__header">Registration</div>
            <Input value={email} setValue={setEmail} type='text' placeholder='Enter your email' />
            <Input value={password} setValue={setPassword} type='password' placeholder='Enter your password' />
            <button className="authorization__btn" onClick={() => registration(email, password)}>Register</button>
        </div>
    );
};

export default Registration;
