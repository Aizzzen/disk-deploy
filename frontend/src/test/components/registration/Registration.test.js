import {render, screen} from "@testing-library/react";
import Registration from "../../../components/registration/Registration";
import userEvent from "@testing-library/user-event";

describe('Registration component tests',  () => {

    test('Main tests', () => {
        render(<Registration/>)
        const containerDiv = screen.getByTestId('register-page')
        const headerDiv = screen.getByText(/registration/i)
        const emailInput = screen.getByPlaceholderText(/Enter your email/i)
        const passInput = screen.getByPlaceholderText(/Enter your password/i)
        const btn = screen.getByRole('button')

        expect(containerDiv).toBeInTheDocument()
        expect(headerDiv).toBeInTheDocument()
        expect(emailInput).toBeInTheDocument()
        expect(passInput).toBeInTheDocument()
        expect(btn).toBeInTheDocument()

        expect(btn).toMatchSnapshot()
        expect(headerDiv).toMatchSnapshot()
        expect(emailInput).toMatchSnapshot()
        expect(passInput).toMatchSnapshot()
        expect(btn).toMatchSnapshot()
    })

    test('Registration events', () => {
        render(<Registration/>)
        const input = screen.getByPlaceholderText(/Enter your email/i)
        expect(input).toContainHTML('')
        userEvent.type(input, 'user@mail.ru')
        expect(input).toContainHTML('user@mail.ru')

        const passInput = screen.getByPlaceholderText(/Enter your password/i)
        expect(passInput).toContainHTML('')
        userEvent.type(passInput, '123456')
        expect(passInput).toContainHTML('123456')
    })

});
