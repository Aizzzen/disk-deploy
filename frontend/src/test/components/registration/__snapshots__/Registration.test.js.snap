// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component tests Main tests 1`] = `
<button
  class="authorization__btn"
>
  Register
</button>
`;

exports[`Registration component tests Main tests 2`] = `
<div
  class="authorization__header"
>
  Registration
</div>
`;

exports[`Registration component tests Main tests 3`] = `
<input
  placeholder="Enter your email"
  type="text"
  value=""
/>
`;

exports[`Registration component tests Main tests 4`] = `
<input
  placeholder="Enter your password"
  type="password"
  value=""
/>
`;

exports[`Registration component tests Main tests 5`] = `
<button
  class="authorization__btn"
>
  Register
</button>
`;
